WordPress/Classes/ViewRelated/System/Action Sheet/DrawerPresentationController.swift

```diff
@@ -21,13 +21,27 @@ public enum DrawerHeight {
     case hidden
 }
 
+public enum DrawerWidth {
+    // Fills the whole screen width
+    case maxWidth
+
+    // When in compact mode, fills a percentage of the screen
+    case percentage(CGFloat)
+
+    // Width will be equal to the the content height value
+    case contentWidth(CGFloat)
+}
+
 public protocol DrawerPresentable: AnyObject {
     /// The height of the drawer when it's in the expanded position
     var expandedHeight: DrawerHeight { get }
 
     /// The height of the drawer when it's in the collapsed position
     var collapsedHeight: DrawerHeight { get }
 
+    /// The width of the Drawer in compact screen
+    var width: DrawerWidth { get }
```

Comment:
I am worried about this being too vague and could cause conflicts. What about prefixing it? 

`drawerWidth` ?

Reply:
Done!